src/primitives/transaction.cpp

```diff
@@ -246,6 +246,24 @@ CTransaction& CTransaction::operator=(const CTransaction &tx) {
     return *this;
 }
 
+CAmount CTransaction::GetSaplingValueIn() const
+{
+    if (valueBalance < 0) return 0;
+    if (!MoneyRange(valueBalance)) {
+        throw std::runtime_error("CTransaction::GetSaplingValueIn(): value out of range");
+    }
+    return valueBalance;
```

Comment:
negate here, don't negate for `GetSaplingValueOut`

Reply:
`valueBalance` is the sum of spends minus the sum of outputs, so it is positive if ZEC is entering the transparent value pool. `GetSaplingValueIn` is value entering the transparent value pool, matching `GetJoinSplitValueIn`.

Reply:
oops!